src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1140,9 +1142,280 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
 
             return new BoundCall(node, receiver, method, args, argNames, argRefKinds, isDelegateCall: isDelegateCall,
                         expanded: expanded, invokedAsExtensionMethod: invokedAsExtensionMethod,
-                        argsToParamsOpt: argsToParams, resultKind: LookupResultKind.Viable, binderOpt: this, type: returnType, hasErrors: gotError);
+                        argsToParamsOpt: argsToParams, defaultArgumentsOpt, resultKind: LookupResultKind.Viable, binderOpt: this, type: returnType, hasErrors: gotError);
+        }
+
+#nullable enable
+
+        private static SourceLocation GetCallerLocation(SyntaxNode syntax)
+        {
+            switch (syntax.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                    return new SourceLocation(((InvocationExpressionSyntax)syntax).ArgumentList.OpenParenToken);
+                case SyntaxKind.ObjectCreationExpression:
+                    return new SourceLocation(((ObjectCreationExpressionSyntax)syntax).NewKeyword);
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    return new SourceLocation(((ConstructorInitializerSyntax)syntax).ArgumentList.OpenParenToken);
+                case SyntaxKind.PrimaryConstructorBaseType:
+                    return new SourceLocation(((PrimaryConstructorBaseTypeSyntax)syntax).ArgumentList.OpenParenToken);
+                case SyntaxKind.ElementAccessExpression:
+                    return new SourceLocation(((ElementAccessExpressionSyntax)syntax).ArgumentList.OpenBracketToken);
+                default:
+                    return new SourceLocation(syntax.GetFirstToken());
+            }
```

Comment:
>} [](start = 12, length = 1)

Consider using a switch expression to get the token, then `return new SourceLocation(token);`.